src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs

```diff
@@ -200,19 +200,48 @@ public void BuildPackage(PackCommandRunner runner)
             runner.BuildPackage();
         }
 
-        private string[] GetTargetPathsToAssemblies(IPackTaskRequest<IMSBuildItem> request)
+        private IEnumerable<OutputLibFile> InitLibFiles(IMSBuildItem[] libFiles)
         {
-            if (request.TargetPathsToAssemblies == null)
+            var assemblies = new List<OutputLibFile>();
+            if (libFiles == null)
             {
-                return new string[0];
+                return assemblies;
             }
 
-            return request.TargetPathsToAssemblies
-                .Where(path => path != null)
-                .Select(path => path.Trim())
-                .Where(path => path != string.Empty)
-                .Distinct()
-                .ToArray();
+
+            foreach (var assembly in libFiles)
+            {
+                var finalOutputPath = assembly.GetProperty("FinalOutputPath");
+                var targetPath = assembly.GetProperty("TargetPath");
+                var targetFramework = assembly.GetProperty("TargetFramework");
+
+                if (!File.Exists(finalOutputPath))
+                {
+                    throw new FileNotFoundException(string.Format(Strings.Error_FileNotFound, finalOutputPath));
+                }
+
+                // If target path is not set, default it to the file name. Only satellite DLLs have a special target path
+                // where culture is part of the target path. This condition holds true for files like runtimeconfig.json file
+                // in netcore projects.
+                if (targetPath == null)
+                {
+                    targetPath = Path.GetFileName(finalOutputPath);
+                }
+
+                if (string.IsNullOrEmpty(targetFramework) || NuGetFramework.Parse(targetFramework) == null)
```

Comment:
I think parse always returns a framework, you may want to check ``NuGetFramework.Parse(targetFramework).IsSpecificFramework == false`` which would mean it came back as unsupported or something else that a project should not have.